src/MgvOfferTaking.sol

```diff
@@ -21,55 +21,63 @@ import {MgvHasOffers} from "./MgvHasOffers.sol";
   }
 
   /* # Market Orders */
 
   /* ## Market Order */
   //+clear+
 
-  /* A market order specifies a (`outbound_tkn`,`inbound_tkn`,`tickSpacing`) offer list, a desired total amount of `outbound_tkn` (`takerWants`), and an available total amount of `inbound_tkn` (`takerGives`). It returns four `uint`s: the total amount of `outbound_tkn` received, the total amount of `inbound_tkn` spent, the penalty received by msg.sender (in wei), and the fee paid by the taker (in wei).
+  /* A market order specifies a (`outbound`, `inbound`,`tickSpacing`) offer list, a limit price it is ready to pay (in the form of `maxTick`, the log base 1.0001 of the price), and a volume `fillVolume`. If `fillWants` is true, that volume is the amount of `olKey.outbound` the taker wants to buy. If `fillWants` is false, that volume is the amount of `olKey.inbound` the taker wants to sell.
+  
+  It returns four `uint`s: the total amount of `olKey.outbound` received, the total amount of `olKey.inbound` spent, the penalty received by msg.sender (in wei), and the fee paid by the taker (in wei).
 
-     The `takerGives/takerWants` ratio induces a maximum average ratio that the taker is ready to pay across all offers that will be executed during the market order. It is thus possible to execute an offer with a ratio worse than the initial (`takerGives`/`takerWants`) ratio given as argument to `marketOrder` if some cheaper offers were executed earlier in the market order.
 
-  The market order stops when the ratio has become too high, or when the end of the book has been reached, or:
-  * If `fillWants` is true, the market order stops when `takerWants` units of `outbound_tkn` have been obtained. With `fillWants` set to true, to buy a specific volume of `outbound_tkn` at any ratio, set `takerWants` to the amount desired and `takerGives` to $2^{104}-1$.
-  * If `fillWants` is false, the taker is filling `gives` instead: the market order stops when `takerGives` units of `inbound_tkn` have been sold. With `fillWants` set to false, to sell a specific volume of `inbound_tkn` at any ratio, set `takerGives` to the amount desired and `takerWants` to $0$. */
-  function marketOrderByVolume(OLKey memory olKey, uint takerWants, uint takerGives, bool fillWants)
+  The market order stops when the price exceeds (an approximation of) 1.0001^`maxTick`, or when the end of the book has been reached, or:
+  * If `fillWants` is true, the market order stops when `fillVolume` units of `olKey.outbound` have been obtained. To buy a specific volume of `olKey.outbound` at any price, set `fillWants` to true, set `fillVolume` to volume you want to buy, and set `maxTick` to the `MAX_TICK` constant.
+  * If `fillWants` is false, the market order stops when `fillVolume` units of `olKey.inbound` have been sold. To sell a specific volume of `olKey.inbound` at any price, set `fillWants` to false, set `fillVolume` to the volume you want to sell, and set `maxTick` to the `MAX_TICK` constant. */
```

Comment:
would it make sense to here state how much of the _other_ token you can end up getting/giving based on the maxTick? For fillWants=true it is important the user knows how much takerGives to approve.